openproblems/tasks/spatial_decomposition/datasets/destvi/utils.py

```diff
@@ -226,9 +226,6 @@ def generate_synthetic_dataset(
         st_anndata.uns["key_clustering"] = key_list
         st_anndata.uns["target_list"] = [1] + target_list
 
-    sc_anndata.layers["counts"] = sc_anndata.X.copy()
-    st_anndata.layers["counts"] = st_anndata.X.copy()
-
```

Comment:
Don't understand why this is removed. This would be count layer definition in the dataset function, which I understand is intended.

Reply:
`adata.layers["counts"] = adata.X` is done inside `@loader`, so this is just redundant and confusing, which was @adamgayoso 's original point in #695.

Reply:
I see... it's still there in other data loaders here though i think. I liked it being explicit i guess... easier to track if not hidden in the decorator... but maybe redundant code.

Reply:
I agree it would be more explicit to have the decorator check that there is a counts layer (though I see it's not ideal to have overhead of two versions of the data)

Reply:
The reason we decided to put it in the decorator is to make explicit that `.X` is _required_ to be counts. If each loader sets both `.X` and `.layers['counts']`, then that opens a can of worms over what should be in `.X`. We could, if we wanted, move `.layers['counts'] = .X` into every dataset loader and add a test `assert adata.layers['counts'] is adata.X` but that seems redundant?